src/pages/dashboard/InstitutionFiltering.tsx

```diff
@@ -27,21 +31,60 @@ interface InstitutionFilteringProps {
 
 const InstitutionFiltering: FC<InstitutionFilteringProps> = ({ queryObject, setQueryObject }) => {
   const { t } = useTranslation();
-  const [institutionCheckedList, setInstitutionCheckedList] = useState(initialInstitutionCheckedList);
+  const [isLoading, setIsloading] = useState(false);
+  const [institutionsFromGetFacets, setInstitutionsFromGetFacets] = useState<string[]>(AllDLRInstitutionNames);
+  const [calledAPIOnce, setCalledAPIONCE] = useState(false);
+  const [institutionCheckedList, setInstitutionCheckedList] = useState(
+    initialInstitutionCheckedList(AllDLRInstitutionNames)
+  );
+  const [error, setError] = useState<Error | null>(null);
+  const mountedRef = useRef(true);
 
   useEffect(() => {
-    if (queryObject.institutions.length > 0) {
-      const nextState = AllDLRInstitutionNames.map((institutionName) => ({
-        name: institutionName,
-        isSelected: !!queryObject.institutions.find(
-          (instName) => instName.toLowerCase() === institutionName.toLowerCase()
-        ),
-      }));
-      setInstitutionCheckedList(nextState);
+    return () => {
+      mountedRef.current = false;
+    };
+  }, []);
+
+  useEffect(() => {
+    const updateInstitutionCheckedList = (list: string[]) => {
+      if (queryObject.institutions.length > 0) {
+        const nextState = list.map((institutionName) => ({
+          name: institutionName,
+          isSelected: !!queryObject.institutions.find(
+            (instName) => instName.toLowerCase() === institutionName.toLowerCase()
+          ),
+        }));
+        setInstitutionCheckedList(nextState);
+      } else {
+        setInstitutionCheckedList(initialInstitutionCheckedList(list));
+      }
+    };
+    const generateInstitutionListFromFacets = async () => {
+      setError(null);
```

Comment:
setError kan gjerne også inni try-løkka - bare for konsistent struktur

Reply:
Fikset